new: more adapter housekeeping

aMahanna opened the pull request:
Note: The scope of this PR has widened, but the branch name has not been updated

### What's new
* raises a `ValueError` if the result of `identify_cugraph_node` or `identify_cugraph_edge` is not valid
* adds node/edge level logging via a `tqdm` progress meter
* docstring updates
* new `test_cug_to_adb_invalid_collections` test case
* introduces the optional `weight` parameter to `identify_cugraph_edge`
* new `__insert_adb_docs` method

aMahanna committed:
initial commit

aMahanna committed:
cleanup

aMahanna committed:
new: debug logs on individual node & edge levels

Is this too much?

aMahanna committed:
cleanup

aMahanna committed:
new: pass edge weight value to _identify_cugraph_edge controller method

aMahanna committed:
new: test_cug_to_adb_invalid_collections case

aMahanna committed:
fix: flake8

aMahanna committed:
fix: mypy

aMahanna committed:
fix: tests

aMahanna committed:
update notebook

cw00dw0rd commented:
Discussed in person:

Investigate python-arango logging
Use logging for better error handling as well

aMahanna committed:
cleanup

aMahanna committed:
fix: logger debug

aMahanna committed:
remove: overwrite option in test

aMahanna committed:
Revert "remove: overwrite option in test"

This reverts commit ee5e53c167974c314abd245bcd1fd59596cab4f4.

aMahanna committed:
new: __insert_adb_docs

also: replaces the "weight" DataFrame column name with whatever specified in the **edge_attr** parameter

aMahanna committed:
Merge branch 'master' into feature/raise-on-invalid-collection

cw00dw0rd commented:
Discussed logging levels but good to merge as is. Thank you!

aMahanna committed:
cleanup config: tests

aMahanna committed:
replace: node/edge level debug statements in favor of `tqdm` progress ba…

aMahanna committed:
fix: tqdm as a dependency

aMahanna committed:
pragma: no cover

aMahanna committed:
Update README.md

aMahanna committed:
Update adapter.py

aMahanna committed:
fix: black

cw00dw0rd reviewed (approved):
LGTM!
